src/foolscap/banana.py

```diff
@@ -180,7 +180,7 @@ def initSlicer(self):
         assert tokens.IRootSlicer.providedBy(self.rootSlicer)
 
         itr = self.rootSlicer.slice()
-        next = iter(itr).next
+        next = iter(itr).__next__
```

Comment:
This will fail in 2/3 code; py2 has no `__next__`.

Reply:
This one has as I specifically added a `__next__` to the class. Here is it in pdb.

```
> /home/anand/projects/anvetsu/tahoe/foolscap/src/foolscap/banana.py(184)initSlicer()
-> next = iter(itr).__next__
(Pdb) iter(itr)
<foolscap.storage.UnsafeStorageRootSlicer instance at 0x7fada7fe1e18>
(Pdb) dir(iter(itr))
['__doc__', '__implemented__', '__init__', '__iter__', '__module__', '__next__', '__providedBy__', '__provides__', 'allowStreaming', 'childAborted', 'connectionLost', 'debug', 'describe', 'next', 'objectSentDeferred', 'producingDeferred', 'protocol', 'references', 'registerRefID', 'send', 'sendQueue', 'slice', 'slicerForObject', 'slicerTable', 'streamableInGeneral']
```
I could've done `if six.PY2 ...` here as well. But this works without it for 2 and 3.